js/idex.js

```diff
@@ -0,0 +1,969 @@
+        //         "sellPrecision": 18,
+        //         "amountSell": "206163978911921061732",
+        //         "expires": 10000,
+        //         "nonce": 13489307413,
+        //         "user": "0x9e8ef79316a4a79bbf55a5f9c16b3e068fff65c6" } } ],
+        //   "bids": [ {
+        //       "price": "0.001161865193232242",
+        //       "amount": "854.393661648355",
+        //       "total": "0.992690256787469029",
+        //       "orderHash": "0x2f2baaf982085e4096f9e23e376214885fa74b2939497968e92222716fc2c86d",
+        //       "params": {
+        //         "tokenBuy": "0xb98d4c97425d9908e66e53a6fdf673acca0be986",
+        //         "buySymbol": "ABT",
+        //         "buyPrecision": 18,
+        //         "amountBuy": "854393661648355000000",
+        //         "tokenSell": "0x0000000000000000000000000000000000000000",
+        //         "sellSymbol": "ETH",
+        //         "sellPrecision": 18,
+        //         "amountSell": "992690256787469029",
+        //         "expires": 10000,
+        //         "nonce": 18155189676,
+        //         "user": "0xb631284dd7b74a846af5b37766ceb1f85d53eca4" } } ] }
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'amount');
+    }
+
+    async fetchBalance (params = {}) {
+        const request = {
+            'address': this.walletAddress,
+        };
+        const response = await this.publicPostReturnCompleteBalances (this.extend (request, params));
+        //    { ETH:
+        //         { available: '0.0167',
+        //           onOrders: '0.1533' }
+        //    }
+        const result = {
+            'info': response,
+        };
+        const keys = Object.keys (response);
+        for (let i = 0; i < keys.length; i++) {
+            const currency = keys[i];
+            const balance = response[currency];
+            result[currency] = {
+                'free': this.safeFloat (balance, 'available'),
+                'used': this.safeFloat (balance, 'onOrders'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        this.checkRequiredCredentials ();
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (type === 'limit') {
+            const expires = 100000;
+            const contractAddress = await this.getContractAddress ();
+            let tokenBuy = undefined;
+            let tokenSell = undefined;
+            let amountBuy = undefined;
+            let amountSell = undefined;
+            const quoteAmount = parseFloat (price) * parseFloat (amount);
+            if (side === 'buy') {
+                tokenBuy = market['baseId'];
+                tokenSell = market['quoteId'];
+                amountBuy = this.toWei (amount, 'ether', market['precision']['amount']);
+                amountSell = this.toWei (quoteAmount, 'ether', 18);
```

Comment:
We need to make this line a bit more complicated, namely, we need to convert to ERC20 values using the `decimals` as defined in the corresponding token contracts (`ETH`, `TUSD`, `USDC`, `EURS`, `DAI`). This edit naturally follows from the above comment on supporting all markets.